### PR TITLE
Migration guide for PrimaryScrollController Updates

### PR DESCRIPTION
**Pull request by Piinks**
Adds a migration guide for changes from https://github.com/flutter/flutter/pull/102099

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

File: src/release/breaking-changes/primary-scroll-controller-desktop.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+      return Text('Item $index');
+    }
+  ),
+);
+```
+
+The implementation changes `ScrollView.primary` to be nullable, with the fallback
+decision-making being relocated to the `PrimaryScrollController`.
+When `primary` is null, and no `ScrollController` has been provided, the `ScrollView`
+will look up the `PrimaryScrollController` and instead calls `shouldInherit` to
```
Comment (sfshaza2) [inline src/release/breaking-changes/primary-scroll-controller-desktop.md:53]:
```suggestion
will look up the `PrimaryScrollController` and instead call `shouldInherit` to
```

File: src/release/breaking-changes/primary-scroll-controller-desktop.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+determine if the given `ScrollView` should use the `PrimaryScrollController`.
+
+The new members of the `PrimaryScrollController` class,
+`automaticallyInheritForPlatforms`, and `scrollDirection` are evaluated in
```
Comment (sfshaza2) [inline src/release/breaking-changes/primary-scroll-controller-desktop.md:57]:
```suggestion
`automaticallyInheritForPlatforms` and `scrollDirection`, are evaluated in
```

File: src/release/breaking-changes/primary-scroll-controller-desktop.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+`shouldInherit`, allowing users clarity and control over the
+`PrimaryScrollController`'s behavior.
+
+By default, backwards compatibility is maintained for mobile platforms.
+`PrimaryScrollController.shouldInherit` will return true for vertical
+`ScrollView`s. On desktop, this will return false by default.
```
Comment (sfshaza2) [inline src/release/breaking-changes/primary-scroll-controller-desktop.md:63]:
```suggestion
`PrimaryScrollController.shouldInherit` returns true for vertical
`ScrollView`s. On desktop, this returns false by default.
```

File: src/release/breaking-changes/primary-scroll-controller-desktop.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+To change the default, users can set `ScrollView.primary` true or false to
+explicitly manage the `PrimaryScrollController` for an individual `ScrollView`.
+For behavior across multiple `ScrollView`s, the `PrimaryScrollController` is now
+configurable by setting the specific platform, as well as the scroll direction
+that is preferred for inheritance.
+
+Widgets that use the `PrimaryScrollController` such as the `NestedScrollView`,
+`Scrollbar`, and `DropdownMenuButton` will experience no change to existing
+functionality. Features like the iOS scroll-to-top will also continue to work as
+expected without any migration.
```
Comment (sfshaza2) [inline src/release/breaking-changes/primary-scroll-controller-desktop.md:87]:
```suggestion
Widgets that use the `PrimaryScrollController`, such as `NestedScrollView`,
`Scrollbar`, and `DropdownMenuButton` will experience no change to existing
functionality. Features like the iOS scroll-to-top will also continue to work as
expected without any migration.
```

File: src/release/breaking-changes/primary-scroll-controller-desktop.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+`ScrollAction`s, and `ScrollIntent`s on desktop are the only classes affected by
+this change, requiring migration. By default, the `PrimaryScrollController` is
+used to execute fallback keyboard scrolling `Shortcuts` if the current `Focus` is
+contained within a `Scrollable`. Since displaying more than one `ScrollView`
+side-by-side is common on desktop platforms, it is not possible for
```
Comment (sfshaza2) [inline src/release/breaking-changes/primary-scroll-controller-desktop.md:93]:
```suggestion
side-by-side is common on desktop platforms, it isn't possible for
```

File: src/release/breaking-changes/primary-scroll-controller-desktop.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+Flutter to decide "Which `ScrollView` should be primary in this view and receive
+the keyboard scroll action?"
+
+If more than one `ScrollVie`w were present previous to this change, the same
+assertion (`ScrollController attached to multiple ScrollViews.`) would be thrown.
+Now, on desktop platforms, users will need to specify `primary: true` to
+designate which `ScrollView` is the fallback to receive unhandled keyboard
+`Shortcuts`.
```
Comment (sfshaza2) [inline src/release/breaking-changes/primary-scroll-controller-desktop.md:101]:
```suggestion
If more than one `ScrollView` was present previous to this change, the same
assertion (`ScrollController attached to multiple ScrollViews.`) would be thrown.
Now, on desktop platforms, users need to specify `primary: true` to
designate which `ScrollView` is the fallback to receive unhandled keyboard
`Shortcuts`.
```

**Review by sfshaza2 — COMMENTED** [review]
A few edits, but nothing huge.  :)

**Comment by sfshaza2** [comment]
Also, do you have a version where this was merged?

**Comment by Piinks** [comment]
> do you have a version where this was merged?

It was merged a while ago, but it does not have a build tag yet.

**Comment by domesticmouse** [comment]
> > do you have a version where this was merged?
> 
> It was merged a while ago, but it does not have a build tag yet.

@Piinks can you mark your PRs as Draft until you are ready for us to review and land them?

**Comment by Piinks** [comment]
> can you mark your PRs as Draft until you are ready for us to review and land them?

Why would it be a draft? This change has landed and customers inside of google will need this before it gets a version tag.

> do we have the "landed in version" info?

No. As stated above, it does not have a tag yet.

We used to generate version tags when Flutter rolled into Google, this was reflected in the dev channel. The dev channel is no longer being used, and we only tag versions when we roll to beta. This only happens ~ once a month, but in the meantime, Google customers have this untagged version. Should they not have a migration guide while we wait for a beta release?

Also, other migration guides have been merged for changes that have not even landed yet - so why would this not be ready? This guide does not have a version tag, and the change has not even landed yet: 

- https://github.com/flutter/flutter/pull/97972
- https://github.com/flutter/website/pull/7243/files
- https://docs.flutter.dev/release/breaking-changes/toggleable-active-color

Can you help me understand why that can land but this can't?

**Comment by sfshaza2** [comment]
OK, thanks for clarifying. I didn't realize that this was the result of retiring the dev branch.

**Pull request merged**